crates/driver/src/domain/competition/mod.rs

```diff
@@ -213,23 +184,23 @@ impl Competition {
 
         // Re-simulate the solution on every new block until the deadline ends to make
         // sure we actually submit a working solution close to when the winner
-        // gets picked by the procotol.
-        if let Ok(deadline) = auction.deadline().timeout() {
+        // gets picked by the protocol.
+        if !auction.deadline().remaining().is_zero() {
             let score_ref = &mut score;
             let simulate_on_new_blocks = async move {
                 let mut stream =
                     ethrpc::current_block::into_stream(self.eth.current_block().clone());
                 while let Some(block) = stream.next().await {
                     if let Err(err) = self.simulate_settlement(&settlement).await {
-                        tracing::warn!(block = block.number, ?err, "solution reverts on new block");
+                        observe::winner_voided(block, &err);
                         *score_ref = None;
                         *self.settlement.lock().unwrap() = None;
```

Comment:
I noticed that we are removing the winning settlement, but not replacing it with a runners up that is still valid. Unrelated to this PR, but another optimisation we could make.